travis/ipa_run_test.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+#
+# Authors:
+#     Dinesh Prasanth M K <dmoluguw@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright (C) 2018 Red Hat, Inc.
+# All rights reserved.
+#
+set -e
+
+# Print the version of installed components
+rpm -qa tomcat* pki-* freeipa-* | sort
+
+# Disable IPV6
+sysctl net.ipv6.conf.lo.disable_ipv6=0
+
+# Define constants
+server_password="Secret.123"
+
+# Install IPA-server
+ipa-server-install -U --domain pki.test --realm PKI.TEST -p ${server_password} -a ${server_password} --setup-dns --setup-kra --auto-forwarders
+
+# Restart httpd
+systemctl restart httpd.service
+
+# Test whether IPA server is reachable
+echo ${server_password} | kinit admin && ipa ping
+
+# Setup environment to run tests
+cp -r /etc/ipa/* ~/.ipa/
+echo ${server_password} > ~/.ipa/.dmpw
+echo 'wait_for_dns=5' >> ~/.ipa/default.conf
+
+# Make a list of IPA tests to run
+cert_test_file_loc=""
+for test_files in ${test_set}; do
+    cert_test_file_loc="${cert_test_file_loc} test_xmlrpc/${test_files}"
+done
+echo "Following IPA tests are scheduled to run: "
+echo ${cert_test_file_loc}
+
+# Run ipa-tests
+ipa-run-tests-3 \
+--ignore test_integration \
+--ignore test_webui \
+--ignore test_ipapython/test_keyring.py \
+-k-test_dns_soa \
+--verbose \
+${cert_test_file_loc}
+
+# Uninstall ipa-server
+ipa-server-install --uninstall -U
+
+# Cleanup and pack the logs
+LOGS_TAR_NAME="var_log.tar"
+journalctl -b --no-pager > systemd_journal.log
+tar --ignore-failed-read -cvf /tmp/${LOGS_TAR_NAME} \
+    /var/log/dirsrv \
+    /var/log/httpd \
+    /var/log/ipa* \
+    /var/log/krb5kdc.log \
+    /var/log/pki \
+    systemd_journal.log
```

Comment:
To simplify troubleshooting let's rename systemd journal to *.txt and upload it separately. That way people can view the journal directly on transfer.sh without having to download and extract the tarball.